sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -1144,8 +1143,8 @@ class CudfRegexTranspiler(mode: RegexMode) {
         case 'z' if mode == RegexSplitMode =>
```

Comment:
We left this split case for `\z`, but the error message just claims that `\z` is not supported on GPU. We should either remove this case, or clarify the error messaging.

Reply:
Updated.